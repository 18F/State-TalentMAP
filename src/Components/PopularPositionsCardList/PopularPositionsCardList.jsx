import React from 'react';
import ResultsCondensedCard from '../ResultsCondensedCard';
import { POSITION_DETAILS_ARRAY } from '../../Constants/PropTypes';

const shortid = require('shortid');

const PopularPositionsCardList = ({ positions }) => {
  const positionList = positions.slice().map(p => (
    <div key={shortid.generate()} className="usa-width-one-third condensed-card">
      <ResultsCondensedCard type="popular" result={p} />
    </div>
  ));
  return (
    <div className="usa-grid-full condensed-card-popular">
      {positionList}
    </div>
  );
};

PopularPositionsCardList.propTypes = {
   // TODO next round - add prop types
  positions: POSITION_DETAILS_ARRAY,
};

PopularPositionsCardList.defaultProps = {
  positions: [{}, {}, {}], // TODO remove and only use real data
};

export default PopularPositionsCardList;
