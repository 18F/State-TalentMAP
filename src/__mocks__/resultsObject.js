const resultsObject = {
  count: 2,
  results: [
    { id: 6,
      grade: '05',
      skill: 'OFFICE MANAGEMENT (9017)',
      bureau: '150000',
      organization: 'YAOUNDE CAMEROON (YAOUNDE)',
      position_number: '00025003',
      is_overseas: true,
      create_date: '2006-09-20',
      update_date: '2017-06-08',
      bid_statistics: [
        {
          id: 1,
          total_bids: 3,
          in_grade: 0,
          at_skill: 0,
          in_grade_at_skill: 0,
          bidcycle: 'Demo BidCycle 2018-01-12 15:11:38.004793',
        },
      ],
      description: {
        content: 'The quick brown fox jumped over the lazy dog.',
        date_created: '2018-01-12T15:09:22.952226Z',
      },
      post: { id: 162, obc_id: 162, tour_of_duty: '2YRR', code: 'LT6000000', location: 'MASERU, LESOTHO', cost_of_living_adjustment: 0, differential_rate: 15, danger_pay: 0, rest_relaxation_point: 'London', has_consumable_allowance: false, has_service_needs_differential: false },
      languages: [
        { id: 1, language: 'French (FR)', reading_proficiency: '2', spoken_proficiency: '2', representation: 'French (FR) 2/2' },
        { id: 1, language: 'German (GM)', reading_proficiency: '2', spoken_proficiency: '2', representation: 'German (GM) 2/2' },
      ],
      title: 'Position - First',
    },
    { id: 60,
      grade: '03',
      skill: 'OFFICE MANAGEMENT (9017)',
      bureau: '150000',
      organization: 'YAOUNDE CAMEROON (YAOUNDE)',
      position_number: '00025003',
      is_overseas: true,
      create_date: '2006-09-20',
      update_date: '2017-06-08',
      bid_statistics: [
        {
          id: 1,
          total_bids: 3,
          in_grade: 0,
          at_skill: 0,
          in_grade_at_skill: 0,
          bidcycle: 'Demo BidCycle 2018-01-12 15:11:38.004793',
        },
      ],
      post: null,
      languages: [],
      title: 'Position - Second',
    },
  ],
};

export default resultsObject;
