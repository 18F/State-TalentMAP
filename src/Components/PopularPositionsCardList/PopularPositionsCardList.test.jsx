import { shallow } from 'enzyme';
import React from 'react';
import toJSON from 'enzyme-to-json';
import PopularPositionsCardList from './PopularPositionsCardList';
import resultsObject from '../../__mocks__/resultsObject';

describe('PopularPositionsCardListComponent', () => {
  it('is defined', () => {
    const wrapper = shallow(
      <PopularPositionsCardList
        positions={resultsObject.results}
        toggleFavorite={() => {}}
        userProfileFavoritePositionIsLoading={false}
        userProfileFavoritePositionHasErrored={false}
      />,
    );
    expect(wrapper).toBeDefined();
  });

  it('can receive props', () => {
    const wrapper = shallow(
      <PopularPositionsCardList
        positions={resultsObject.results}
        toggleFavorite={() => {}}
        userProfileFavoritePositionIsLoading={false}
        userProfileFavoritePositionHasErrored={false}
      />,
    );
    expect(wrapper.instance().props.positions[0].id).toBe(resultsObject.results[0].id);
  });

  it('can change className based on isLoading', () => {
    const wrapper = shallow(
      <PopularPositionsCardList
        positions={resultsObject.results}
        toggleFavorite={() => {}}
        userProfileFavoritePositionIsLoading={false}
        userProfileFavoritePositionHasErrored={false}
        isLoading
      />,
    );
    expect(wrapper.find('.results-loading')).toBeDefined();
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <PopularPositionsCardList
        positions={resultsObject.results}
        toggleFavorite={() => {}}
        userProfileFavoritePositionIsLoading={false}
        userProfileFavoritePositionHasErrored={false}
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
