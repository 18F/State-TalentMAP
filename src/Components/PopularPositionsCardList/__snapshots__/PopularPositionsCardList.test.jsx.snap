// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopularPositionsCardListComponent matches snapshot 1`] = `
<div
  className="usa-grid-full condensed-card-popular"
>
  <div
    className="usa-width-one-third condensed-card"
  >
    <ResultsCondensedCard
      favorites={Array []}
      position={
        Object {
          "bureau": "150000",
          "create_date": "2006-09-20",
          "grade": "05",
          "id": 6,
          "is_overseas": true,
          "languages": Array [
            Object {
              "id": 1,
              "language": "French (FR)",
              "reading_proficiency": "2",
              "representation": "French (FR) 2/2",
              "spoken_proficiency": "2",
            },
            Object {
              "id": 1,
              "language": "German (GM)",
              "reading_proficiency": "2",
              "representation": "German (GM) 2/2",
              "spoken_proficiency": "2",
            },
          ],
          "organization": "YAOUNDE CAMEROON (YAOUNDE)",
          "position_number": "00025003",
          "post": Object {
            "code": "LT6000000",
            "cost_of_living_adjustment": 0,
            "danger_pay": 0,
            "differential_rate": 15,
            "has_consumable_allowance": false,
            "has_service_needs_differential": false,
            "id": 162,
            "location": "MASERU, LESOTHO",
            "rest_relaxation_point": "London",
            "tour_of_duty": "2YRR",
          },
          "skill": "OFFICE MANAGEMENT (9017)",
          "update_date": "2017-06-08",
        }
      }
      toggleFavorite={[Function]}
      type="popular"
      userProfileFavoritePositionHasErrored={false}
      userProfileFavoritePositionIsLoading={false}
    />
  </div>
  <div
    className="usa-width-one-third condensed-card"
  >
    <ResultsCondensedCard
      favorites={Array []}
      position={
        Object {
          "bureau": "150000",
          "create_date": "2006-09-20",
          "grade": "03",
          "id": 60,
          "is_overseas": true,
          "languages": Array [],
          "organization": "YAOUNDE CAMEROON (YAOUNDE)",
          "position_number": "00025003",
          "post": null,
          "skill": "OFFICE MANAGEMENT (9017)",
          "update_date": "2017-06-08",
        }
      }
      toggleFavorite={[Function]}
      type="popular"
      userProfileFavoritePositionHasErrored={false}
      userProfileFavoritePositionIsLoading={false}
    />
  </div>
</div>
`;
